src/averageAges.js

```diff
@@ -52,7 +56,14 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let child = people.filter(baby => !onlyWithSon
+    ? people.some(mom => mom.name === baby.mother)
+    : people.some(mom => mom.name === baby.mother) && baby.sex === 'm');
+
+  child = child.map(person =>
```

Comment:
don't reuse one variable for different things better to create two constant in this case